src/pages/incident/incident-list.js

```diff
@@ -112,6 +113,29 @@ export function IncidentListPage() {
     });
   }, []);
 
+  async function fetchIncidents(params) {
+    try {
+      const res = await getIncidentsWithParams(params);
+      const personIds = [];
+      res.data.forEach((item) => {
+        if (!personIds.includes(item.communicator_id)) {
+          personIds.push(item.communicator_id);
+        }
+      });
+      const persons = (await getResponder(personIds.toString())).data;
```

Comment:
This should be done server side by joining the incident and responder tables   see https://postgrest.org/en/stable/api.html#resource-embedding